Fix stack installation in alpine

coord-e opened the pull request:
Small PR to fix installation using installation script in alpine linux. `do_alpine_install` is commented out in 87ff1220c579ec03cff8874d6ee59e87b1b0aec0 but it still occurred in the installation. That caused an error as shown below.

```
/workdir # ./etc/scripts/get-stack.sh
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/community/x86_64/APKINDEX.tar.gz
(1/2) Installing libbz2 (1.0.6-r7)
(2/2) Installing perl (5.28.2-r1)
Executing busybox-1.30.1-r2.trigger
OK: 44 MiB in 16 packages
Detected Linux distribution: alpine
./etc/scripts/get-stack.sh: line 465: do_alpine_install: not found
```

This PR fixes this by simply skipping the branch for alpine in `do_distro` and make the script try "generic" installation. I tested this change by running `./etc/scripts/get-stack.sh` in a docker container of `alpine:3.11` image and confirmed that the installation finishes successfully with a warning "there is no guarantee that 'stack' will work at all!..." as expected.

Please include the following checklist in your PR:

* [x] Any changes that could be relevant to users have been recorded in the ChangeLog.md
  * I'm not sure whether this should be recorded in changelog
* [x] The documentation has been updated, if necessary.
  * I this there is no need for this

snoyberg commented:
I've asked @borsboom to review. One comment though: I think this should target the `stable` branch instead.

borsboom commented:
Ack, I guess I was a bit overzealous when I was commenting out code for no-longer-supported platforms.  Thanks!

I think a better fix would be to un-comment the `do_alpine_install` function, rather than fallback to `do_sloppy_install`.

And, as @snoyberg suggested, base this on `stable` rather than `master`.

coord-e commented:
@snoyberg @borsboom Thanks for your reviews! I've fixed the base branch to `stable`, and changed my fix to un-comment `do_alpine_install` with small modifications.

I'm willing to write a pull-req that enables `shellcheck` on the CI workflow to prevent this kind of issue. I found #3675 but it has suddenly closed, is it OK if I try this?

borsboom commented:
I'd be in favour of running `shellcheck` from CI.  I should have done so myself before I committed my changes.
